runner/runners/local_output.go

```diff
@@ -84,58 +82,123 @@ func (s *localOutputCreator) Create(id string) (runner.Output, error) {
-	s.mutex.Lock()
-	defer s.mutex.Unlock()
-
 	clientHtml :=
 		`<html>
-<script type="text/javascript">
-  var prevLength = 0
-  var resourceId = ''
-  checkAtBottom = function() {
-    //scrolling: http://stackoverflow.com/a/22394544
-    var scrollTop = (document.documentElement && document.documentElement.scrollTop) || document.body.scrollTop;
-    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
-    return (scrollTop + window.innerHeight) >= scrollHeight;
-  }
-  gotoBottom = function() {
-    var scrollHeight = (document.documentElement && document.documentElement.scrollHeight) || document.body.scrollHeight;
-    var scrollLeft = (document.documentElement && document.documentElement.scrollLeft) || document.body.scrollLeft;
-    window.scrollTo(scrollLeft, scrollHeight);
-  }
-  sendRequest = function() {
-    var xhr = new XMLHttpRequest();
-    xhr.onreadystatechange = function () {
-      var DONE=4, OK=200;
-      if (xhr.readyState === DONE && xhr.status == OK) {
-        var id = xhr.getResponseHeader('X-Resource-Id')
-        var txt = xhr.responseText.substring(prevLength);
-        prevLength = xhr.responseText.length
-        if (resourceId == '')
-          resourceId = id
-        if (id != resourceId) {
-          txt = Date() + ': Underlying resource changed! Quitting.'
-        }
-        var wasAtBottom = checkAtBottom()
-        var div = document.getElementById("output");
-        var content = document.createTextNode(txt);
-        div.appendChild(content);
-        if (wasAtBottom)
-          gotoBottom()
-        if (id != resourceId) {
-          throw new Error('ResourceId mismatch.')
-        }
-      }
-    }
-    //TODO: request range to get delta.
-    xhr.open("GET", location.href + (location.search=="" ? "?" : "&") + "content=true");
-    xhr.send();
-  };
-  sendRequest()
-  setInterval(sendRequest, 5000)
-</script>
-<body><div id="output" style="white-space: pre-wrap"></div></body>
-</html>
+		<script type="text/javascript">
+			let resourceId = "";
+			const sleep = ms => new Promise(resolve => setTimeout(resolve, ms));
+			const checkAtBottom = () => {
+				//scrolling: http://stackoverflow.com/a/22394544
+				let scrollTop =
+					(document.documentElement && document.documentElement.scrollTop) ||
+					document.body.scrollTop;
+				let scrollHeight =
+					(document.documentElement && document.documentElement.scrollHeight) ||
+					document.body.scrollHeight;
+				return scrollTop + window.innerHeight >= scrollHeight;
+			};
+			const gotoBottom = () => {
+				let scrollHeight =
+					(document.documentElement && document.documentElement.scrollHeight) ||
+					document.body.scrollHeight;
+				let scrollLeft =
+					(document.documentElement && document.documentElement.scrollLeft) ||
+					document.body.scrollLeft;
+				window.scrollTo(scrollLeft, scrollHeight);
+			};
+			const getLength = resp => {
```

Comment:
for those of us not familiar with javascript... can you comment what this is the length of.... it also appears to return an array not a scalar.

Reply:
Done